core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -201,18 +201,35 @@ private[spark] class ExecutorAllocationManager(
   }
 
   /**
-   * If the add time has expired, request new executors and refresh the add time.
-   * If the remove time for an existing executor has expired, kill the executor.
+   * The number of executors we would have if the cluster manager were to fulfill all our requests.
+   */
+  private def targetNumExecutors(): Int =
+    numExecutorsPending + executorIds.size
+
+  /**
+   * The maximum number of executors we would need under the current load to satisfy all running
+   * and pending tasks.
+   */
+  private def maxNumExecutorsNeeded(): Int = {
+    // The maximum number of executors we need under the current load is the total number of
+    // running or pending tasks, divided by the full task capacity of each executor, rounded up.
```

Comment:
this comment is a little redundant given the javadocs. I would just add `rounded up` to the end of the javadoc itself.